packages/valory/skills/transaction_settlement_abci/rounds.py

```diff
@@ -357,6 +360,12 @@ class SelectKeeperTransactionSubmissionARound(CollectSameUntilThresholdRound):
     no_majority_event = Event.NO_MAJORITY
     collection_key = get_name(SynchronizedData.participant_to_selection)
     selection_key = get_name(SynchronizedData.keepers)
+    required_class_attributes: Tuple[str, ...] = (
+        "done_event",
+        "no_majority_event",
+        "collection_key",
+        "selection_key",
+    )
```

Comment:
Why we need this here? doesn't it come from CollectSameUntilThresholdRound?

Reply:
This is to override `required_class_attributes` in this class because we also override `end_block()` here. Since, there is no need for the `none_event` here, we set a new tuple in `required_class_attributes` with "none_event" removed.